mailpoet/lib/Automation/Engine/Control/StepHandler.php

```diff
@@ -77,6 +77,11 @@ public function __construct(
     $this->registry = $registry;
   }
 
+  public function resetStepRunners(): void {
+    $this->stepRunners = [];
+    $this->addStepRunner(Step::TYPE_ACTION, $this->actionStepRunner);
+  }
```

Comment:
Seems like the service doesn't know which step runners should be included "by default". It might be better to add a getter and a setter and use it like this in the tests:

```php
$this->originalStepRunners = $this->testee->getStepRunners();
// ...
$this->testee->setStepRunners($this->originalStepRunners);
```

Reply:
Addressed and squashed in [c61c118](https://github.com/mailpoet/mailpoet/pull/4452/commits/c61c118590447e5a1909053290c60a84002a51ee)